app/entities/account.go

```diff
@@ -0,0 +1,119 @@
+package entities
+
+import (
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Represents a user account in the digital bank.
+type Account struct {
+	id         string
+	name       string
+	cpf        string
+	secret     string
+	balance    int64
+	created_at time.Time
+}
+
+// constructor
+func NewAccount(id, cpf, name, secret string) (*Account, error) {
```

Comment:
Temos um padrão de evitar usar ponteiro e utilizar apenas em casos onde há necessidade. Uma entidade ser ponteiro pode trazer algumas preocupações pra gente. Exemplo:

```
var account *entities.Account
// como account é nil neste caso, este código vai dar panic
cpf := account.GetCPF()
```

Reply:
Neste caso eu acho mais adequado remover o ponteiro do account retornado e, caso vc precise do endereço desta entidade, vc pode utilizar o prefixo `&` para acessá-lo.
```suggestion
func NewAccount(id, cpf, name, secret string) (Account, error) {
```

Também vai precisar mudar a decleração da variável account removendo o `&`:
```
// era &Account{ 
account := Account{
		id:         id,
		cpf:        cpf,
		name:       name,
		secret:     secret,
		balance:    0,
		created_at: time.Now(),
	}